core/divelist.c

```diff
@@ -85,6 +85,38 @@ static int active_o2(const struct dive *dive, const struct divecomputer *dc, dur
 	return get_o2(gas);
 }
 
+static int get_sample_o2(const struct dive *dive, const struct divecomputer *dc, const struct sample *sample)
+{
+	int po2i, po2f, po2;
+	bool trueo2 = false;
+	struct sample *psample = sample - 1;
+	if ((dc->divemode == CCR || dc->divemode == PSCR) && sample->o2sensor[0].mbar) {			// if dive computer has o2 sensor(s) (CCR & PSCR)
+		po2i = psample->o2sensor[0].mbar;
+		po2f = sample->o2sensor[0].mbar;	// then use data from the first o2 sensor
+		po2 = (po2f + po2i) / 2;
+		trueo2 = true;
+	}
+	if ((dc->divemode == CCR) && (sample->setpoint.mbar > 0) && (!trueo2)) {
+		po2 = MIN((int) sample->setpoint.mbar,
+			   depth_to_mbar(sample->depth.mm, dive));
+		trueo2 = true;
+	}
+	if (!trueo2) {
```

Comment:
reading through these three if statements... why isn't this
```
if ((dc->divemode == CCR || dc->divemode == PSCR) && sample->o2sensor[0].mbar) {
...
} else if ((dc->divemode == CCR) && (sample->setpoint.mbar > 0) {
...
} else {
...
}
```
which means we wouldn't need the trueo2 variable and the whole flow would seem a bit more "normal"?

Ahh, I see, this is just existing code that you moved. The question remains :)

Reply:
Cleaned up the logic